### PR TITLE
API-5974 - initial openapi unifier

### PR DESCRIPTION
**Pull request by apeirats-va**

### CODE REVIEW

File: conformance-unifier/pom.xml [hunk 1]
```diff
@@ -11,13 +11,14 @@
   <packaging>jar</packaging>
   <properties>
     <fhir-resources.version>5.0.17</fhir-resources.version>
-    <informational.version>2.0.1</informational.version>
+    <informational.version>2.0.2-SNAPSHOT</informational.version>
```
Comment (apeirats-va) [inline conformance-unifier/pom.xml:14]: I'll use latest release when other PR merges.

**Comment by beaugrantham-va** [comment]
I attempted to unify DQ with Community Care, just to see what happens, and got a NPE merging `servers`. I don't know if `servers` is a required section. Just a note.

**Comment by beaugrantham-va** [comment]
Recommend adding to openapi runs to `run-local.sh` for local testing.

File: conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java [hunk 2]
```diff
@@ -96,6 +106,21 @@ private void unifyMetadata(
         objectName, metadataMap, metadataTransformer.apply(metadataList), "application/fhir+json");
   }
 
+  private void unifyOpenapi(
+      final String objectName, final List<String> urlList, final Map<String, String> metadataMap) {
+    List<O> metadataList = new ArrayList<>();
```
Comment (beaugrantham-va) [inline conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java:111]: This is not a list of metadatas.

File: conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java [hunk 3]
```diff
@@ -96,6 +106,21 @@ private void unifyMetadata(
         objectName, metadataMap, metadataTransformer.apply(metadataList), "application/fhir+json");
   }
 
+  private void unifyOpenapi(
+      final String objectName, final List<String> urlList, final Map<String, String> metadataMap) {
```
Comment (beaugrantham-va) [inline conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java:110]: This is not a map of metadatas.

File: conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java (hunk 2 above)
Comment (beaugrantham-va) [inline conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java:111]: This is not a list of metadatas.

**Comment by beaugrantham-va** [comment]
Similar to the health-apis-informational-endpoints repo, I'd be glad to have another team closer to the APIs own this project.

**Comment by apeirats-va** [comment]
> I attempted to unify DQ with Community Care, just to see what happens, and got a NPE merging `servers`. I don't know if `servers` is a required section. Just a note.

I'll throw in a nullcheck.

Keep in mind this implementation is not a truly generic openapi unifier. This is meant to unify openapi definitions for separate microservices serving the same base endpoint behind the scenes. In this case, both Data Query and Smart Cards are served from `/services/fhir/...`, so I assume the `server` definitions would match. I don't see a need (not yet at least) to combine to truly separate Lighthouse APIs.

A case could be made for adding support for _full_ urls (`servers` section empty, but full URLs published under `paths`). We just don't actually do that in our current definitions.

File: conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java (hunk 2 above)
Comment (beaugrantham-va) [reply conformance-unifier/src/main/java/gov/va/api/health/conformance/unifier/fhir/BaseUnifierService.java:111]: I was actually referencing line 111 (`metadataList`).

The variable you changed (`metadataMap`) was actually correct-ish. It's referring to s3 metadata associated with the unified artifact. It's a bit overloaded here so easy to get confused. Take a look at the Javadoc on the methods you copied. I recommend you replicate it on your method.

**Review by beaugrantham-va — APPROVED** [review]

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**